public/app/features/correlations/Forms/TransformationsEditor.tsx

```diff
@@ -0,0 +1,318 @@
+import { css } from '@emotion/css';
+import { compact, fill } from 'lodash';
+import React, { useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+
+import { GrafanaTheme2, SupportedTransformationType } from '@grafana/data';
+import { Stack } from '@grafana/experimental';
+import {
+  Button,
+  Field,
+  FieldArray,
+  Icon,
+  IconButton,
+  Input,
+  InputControl,
+  Label,
+  Select,
+  Tooltip,
+  useStyles2,
+} from '@grafana/ui';
+
+type Props = { readOnly: boolean };
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  heading: css`
+    font-size: ${theme.typography.h5.fontSize};
+    font-weight: ${theme.typography.fontWeightRegular};
+  `,
+  // set fixed position from the top instead of centring as the container
+  // may get bigger when the for is invalid
+  removeButton: css`
+    margin-top: 25px;
+  `,
+});
+
+export const TransformationsEditor = (props: Props) => {
+  const { control, formState, register, setValue, watch, getValues } = useFormContext();
+  const { readOnly } = props;
+  const [keptVals, setKeptVals] = useState<Array<{ expression?: string; mapValue?: string }>>([]);
+
+  const styles = useStyles2(getStyles);
+
+  const transformOptions = getTransformOptions();
+  return (
+    <>
+      <input type="hidden" {...register('id')} />
+      <FieldArray name="config.transformations" control={control}>
+        {({ fields, append, remove }) => (
+          <>
+            <Stack direction="column" alignItems="flex-start">
+              <div className={styles.heading}>Transformations</div>
+              {fields.length === 0 && <div> No transformations defined.</div>}
+              {fields.length > 0 && (
+                <div>
+                  {fields.map((fieldVal, index) => {
+                    return (
+                      <Stack direction="row" key={fieldVal.id} alignItems="top">
+                        <Field
+                          label={
+                            <Stack gap={0.5}>
+                              <Label htmlFor={`config.transformations.${index}.type`}>Type</Label>
```

Comment:
we need to use the same id as `inputId` prop in the `Select` at line 81 otherwise that select is going to be unlabeled.

moreover, we should not use the index here as it causes this label to target all transformations being edited with the same index:

https://user-images.githubusercontent.com/1170767/232199235-5430d621-b380-43e1-8492-40100650e06c.mov


we could use ``htmlFor={`config.transformations.${fieldVal.id}.type`}`` here and ``inputId={`config.transformations.${fieldVal.id}.type`}`` in the select. and by doing that you should be able to leverage this label element to describe the input instead of using an explicit `aria-label` on the select


```suggestion
                              <Label htmlFor={`config.transformations.${fieldVal.id}.type`}>Type</Label>
```